packages/atomic/src/utils/initialization-utils.tsx

```diff
@@ -1,53 +1,136 @@
-import {Engine} from '@coveo/headless';
+import {Controller, Engine} from '@coveo/headless';
 import {ComponentInterface, getElement, h} from '@stencil/core';
 import {i18n} from 'i18next';
 
-export interface InterfaceContext {
+/**
+ * Bindings passed from the `AtomicSearchInterface` to its children components.
+ */
+export interface Bindings {
+  /**
+   * Headless Engine instance.
+   */
   engine: Engine;
+  /**
+   * i18n instance, for localization.
+   */
   i18n: i18n;
 }
 
-export type InitializeEventHandler = (context: InterfaceContext) => void;
+export type InitializeEventHandler = (bindings: Bindings) => void;
 export type InitializeEvent = CustomEvent<InitializeEventHandler>;
 
-export class InitializationError extends Error {
+export class MissingInterfaceParentError extends Error {
   constructor(elementName: string) {
     super(
       `The "${elementName}" element must be the child of a configured "atomic-search-interface" element.`
     );
-    this.name = 'InitializationError';
   }
 }
 
-export function Initialization() {
-  return (component: ComponentInterface, initializeMethod: string) => {
+/**
+ * Necessary interface an Atomic Component must have to initialize itself correctly.
+ */
+export interface AtomicComponentInterface extends ComponentInterface {
+  /**
+   * Bindings passed from the `AtomicSearchInterface` to its children components.
+   */
+  bindings: Bindings;
+  /**
+   * Record of methods earch returning an i18n localized string.
+   */
+  strings?: Record<string, () => string>;
```

Comment:
So each of these returns a method which returns a string? So for example if I wanted to get the "clear" string I would call `this.strings.clear()` and it would return the correct value? Also if so would these values be bound to state correctly? I'm wondering because the `atomic-text` component will work for string values but we'll need another way to get the correct value for labels.

Reply:
Also a small typo you have it as earch instead of each right now in the comment.

Reply:
Yes, you have a @State property with an object of methods returning a string from i18next.t()